File: Problems/0088.Merge-Sorted-Array/MergeSortedArray_test.py

from MergeSortedArray import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
    ("Example 1", [1,2,3,0,0,0], 3, [2,5,6], 3, [1,2,2,3,5,6]),
    ("Example 2", [0], 0, [1], 1, [1]),
    ("Example 3", [2, 0], 1, [1], 1, [1, 2]),
]

@pytest.mark.parametrize(
    "example, nums1, m, nums2, n, ans", 
    deepcopy(TEST_SUITS)
)
def test(example, nums1, m, nums2, n, ans):
    SOL.merge(nums1, m, nums2, n)
    assert  nums1 == ans

@pytest.mark.parametrize(
    "example, nums1, m, nums2, n, ans", 
    deepcopy(TEST_SUITS)
)
def test_v2(example, nums1, m, nums2, n, ans):
    SOL.merge_v2(nums1, m, nums2, n)
    assert  nums1 == ans

File: leetcode/0088.Merge-Sorted-Array/MergeSortedArray_test.go
package leetcode

import (
	"reflect"
	"testing"
)

func Test(t *testing.T) {
	testSuits := []struct {
		name  string
		nums1 []int
		m     int
		nums2 []int
		n     int
		ans   []int
	}{
		{"Example 1", []int{1, 2, 3, 0, 0, 0}, 3, []int{2, 5, 6}, 3, []int{1, 2, 2, 3, 5, 6}},
		{"Example 2", []int{0}, 0, []int{1}, 1, []int{1}},
		{"Example 3", []int{2, 0}, 1, []int{1}, 1, []int{1, 2}},
	}
	for _, test := range testSuits {
		merge(test.nums1, test.m, test.nums2, test.n)

		if !reflect.DeepEqual(test.nums1, test.ans) {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, test.nums1)
		}
	}
}
